### PR TITLE
feat: :sparkles: add sha-256 hashes for csp

### PR DESCRIPTION
**Pull request by climba03003**
Resolve: #294 

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Comment by climba03003** [comment]
Some questions needed to be answer before going.
1. Better name for ```swagger-sha```
2. Webpack will break the sha256 CSP

**Review by fox1t — COMMENTED** [review]
I like the new naming better! 

LGTM

File: README.md [hunk 1]
```diff
@@ -277,6 +277,23 @@ Sometimes you may need to hide a certain route from the documentation, there is
...
+```javascript
+.register(helmet, {
+  contentSecurityPolicy: {
+    directives: {
+      imgSrc: ["'self'", 'data:', 'validator.swagger.io'],
+      scriptSrc: ["'self'"].concat(fastify.swaggerCSP.script),
+      styleSrc: ["'self'", 'https:'].concat(fastify.swaggerCSP.style)
+    }
+  }
+})
```
Comment (mcollina) [inline README.md:294]: Can you please write a test that verify this works correctly?

**Review by mcollina — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (climba03003) [reply README.md:294]: I have created the integration test and update the ```REMEAD.md``` according to the testing.
I do not know if it is the right way for integration test, but it works.

**Review by mcollina — APPROVED** [review]
lgtm

**Comment by laurenzhonauer** [comment]
Hi, im having 2 Problems with this while using 
`{
"fastify": "^3.11.0",
"fastify-helmet": "^5.3.2",
"fastify-swagger": "^3.3.0",
}
`

1. "Cannot read property 'script' of undefined"
When I configure the plugin like shown in the readme (or done in the integration test) im getting this error. For reference here my JS Code:
```Javascript 
const fastify_helmet_1 = require("fastify-helmet");
const fastifySwagger = require("fastify-swagger");

server.register(fastifySwagger);

server.register(fastify_helmet_1.default, (instance) => {
  console.log("SwaggerCSP is: " + instance.swaggerCSP); // This prints "SwaggerCSP is undefined"

  return {
    contentSecurityPolicy: {
      directives: {
        defaultSrc: ["'self'"],
        imgSrc: ["'self'", "data:", "validator.swagger.io"],
        scriptSrc: ["'self'"].concat(instance.swaggerCSP.script),
        styleSrc: ["'self'", "https:"].concat(instance.swaggerCSP.style),
      },
    },
  };
});
```

2. Im not able to apply this config while using typescript
In the readme, helmet is registered like so:
`.register(helmet, instance => {})`
 This is not possible in typescript because of the way the FastifyRegister interface is done.
As far as I can see I am only allowed to pass a plugin and an opts object like 
```Javascript
server.register(helmet, {
  contentSecurityPolicy: {
    directives: {
      defaultSrc: ["'self'"],
      styleSrc: ["'self'", "'unsafe-inline'"],
      scriptSrc: ["'self'", "'unsafe-inline'"],
    },
  },
});
```

If pass a method instead of the config object im getting error highlighting like in the screenshot below
image


It compiles fine btw, only in VSCode i get the hightlighted error.
Am I doing something wrong here, or is this a problem with the type definition of fastify?

**Comment by mcollina** [comment]
I would recommend opening a fresh issue if you have a problem.

**Comment by laurenzhonauer** [comment]
Oh, sorry, I wasnt really sure where to put this. 
Will do! 😄

**Comment by laurenzhonauer** [comment]
> I would recommend opening a fresh issue if you have a problem.

Would it be better to create 2 seperate issues or 1 containing basically what i have written in my comment above?

**Comment by climba03003** [comment]
> 
> 
> > I would recommend opening a fresh issue if you have a problem.
> 
> @mcollina Would it be better to create 2 seperate issues or 1 containing basically what i have written in my comment above?

I recommend you to update all package once. This feature is added in [`3.4.0`](https://github.com/fastify/fastify-swagger/tree/v3.4.0) and you are using `3.3.0` version of `fastify-swagger`.

**Comment by laurenzhonauer** [comment]
The missing swaggerCSP object was solved through the update. It works from version ` 3.4.0` onwards.

The typescript problem actually only appears while using the `import * as helmet from "fastify-helmet" ` syntax. 
I opened [this issue](https://github.com/fastify/fastify-swagger/issues/485) for that

**Pull request merged**